### PR TITLE
🧹 remove unneeded logic to transform thebe options and improve naming

### PR DESCRIPTION
**Pull request by stevejpurves**
now that `myst-frontmatter` is doing more of the thebe options transformation work the logic in here to go from frontmatter to workable thebe options can be reduced.

UPDATE: this pr now contains more refactoring around the thebe providers and relies on the loaders to resolve projects where needed, leaving the ProjectProvider to provide a useful default and avoid excessive drilling.

tested on book and article theme, with single and multi-project sites

### CODE REVIEW

**Comment by stevejpurves** [comment]
need https://github.com/executablebooks/mystmd/pull/788 to come in and a `myst-frontmatter` release

File: packages/jupyter/package.json [hunk 1]
```diff
@@ -49,5 +49,8 @@
     "@types/react-dom": "^16.8 || ^17.0 || ^18.0",
     "react": "^16.8 || ^17.0 || ^18.0",
     "react-dom": "^16.8 || ^17.0 || ^18.0"
+  },
+  "devDependencies": {
+    "vitest": "^0.34.6"
```
Comment (rowanc1) [inline packages/jupyter/package.json:54]: This is defined in the package and shouldn't need to be pulled in here.

File: packages/jupyter/package.json [hunk 2]
```diff
@@ -29,11 +29,11 @@
...
     "classnames": "^2.3.2",
-    "myst-common": "^1.1.16",
-    "myst-config": "^1.1.16",
-    "myst-frontmatter": "^1.1.16",
+    "myst-common": "^1.1.17",
+    "myst-config": "^1.1.17",
+    "myst-frontmatter": "^1.1.17",
     "myst-spec": "^0.0.4",
-    "myst-spec-ext": "^1.1.16",
+    "myst-spec-ext": "^1.1.17",
```
Comment (rowanc1) [inline packages/jupyter/package.json:36]: Can you ensure all other mystmd deps are also bumped! For example #286 as a template.

**Review by rowanc1 — COMMENTED** [review]
Few comments about dependencies.

File: packages/jupyter/package.json (hunk 1 above)
Comment (stevejpurves) [reply packages/jupyter/package.json:54]: linter was complaining at some point will revert

**Pull request merged**